llama-index-integrations/llms/llama-index-llms-bedrock-converse/llama_index/llms/bedrock_converse/utils.py

```diff
@@ -0,0 +1,228 @@
+import logging
+from typing import Any, Callable, Dict, List, Sequence, Tuple
+from tenacity import (
+    before_sleep_log,
+    retry,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_exponential,
+)
+
+from llama_index.core.base.llms.types import ChatMessage, ChatResponse, MessageRole
+
+
+logger = logging.getLogger(__name__)
+
+HUMAN_PREFIX = "\n\nHuman:"
+ASSISTANT_PREFIX = "\n\nAssistant:"
+
+FUNCTION_CALLING_MODELS = {
+    "anthropic.claude-3-sonnet-20240229-v1:0": 200000,
+    "anthropic.claude-3-haiku-20240307-v1:0": 200000,
+    "anthropic.claude-3-opus-20240229-v1:0": 200000,
+    "cohere.command-r-plus-v1:0": 128000,
+    "mistral.mistral-large-2402-v1:0": 32000,
```

Comment:
does converse support all bedrock models? I noticed that the interface is way easier to managed compared to the current bedrock class, it would be nice if users could use any bedrock model here (and disable function calling where appropriate)

If not, thats cool too

Reply:
Converse supports some LLMs that don't have function calling. However, I've setup this class to inherit from `FunctionCallingLLM`. How do you suggest handling these exceptions of LLMs that don't have function calling?

Reply:
[This commit](https://github.com/run-llama/llama_index/pull/14055/commits/1b910f3aa12958b6ba87f1d5f992a635235e264c) implements a similar behavior to that Mistral integration, allowing for LLMs that don't support function calling but that are in the Converse API ✅

Reply:
Thank you!